include/experimental/__p1673_bits/conjugated.hpp

```diff
@@ -44,135 +44,92 @@
-  return x * T2(y);
-}
-
-template<class Accessor, class T>
+template<class Accessor>
 class accessor_conjugate {
 private:
-  using size_type = typename extents<>::size_type;
-public:
-  using element_type  = typename Accessor::element_type;
-  using pointer       = typename Accessor::pointer;
-  using reference     = typename Accessor::reference;
-  using offset_policy = typename Accessor::offset_policy;
-
-  accessor_conjugate() = default;
+  Accessor accessor_;
 
-  accessor_conjugate(Accessor a) : acc(a) {}
+  using accessor_value_type =
+    std::remove_cv_t<typename Accessor::element_type>;
+  static constexpr bool accessor_value_is_arith =
+    std::is_arithmetic_v<accessor_value_type>;
 
-  reference access(pointer p, size_type i) const noexcept {
-    return reference(acc.access(p, i));
-  }
-
-  typename offset_policy::pointer offset(pointer p, size_type i) const noexcept {
-    return acc.offset(p, i);
-  }
-
-  element_type* decay(pointer p) const noexcept {
-    return acc.decay(p);
-  }
-private:
-  Accessor acc;
-};
+public:
+  using reference =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::reference,
+      conjugated_scalar<typename Accessor::reference, accessor_value_type>>;
 
-template<class Accessor, class T>
-class accessor_conjugate<Accessor, std::complex<T>> {
 private:
-  using size_type = typename extents<>::size_type;
+  using pre_const_element_type =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::element_type,
+      typename reference::value_type>;
+  
 public:
-  // FIXME If BLAS functions want to strip off accessor_conjugate for
-  // optimization, they will need a way to work with th underlying
-  // Accessor (which may not be the default one).
-
-  using element_type  = typename Accessor::element_type;
+  using element_type  = std::add_const_t<pre_const_element_type>;
   using pointer       = typename Accessor::pointer;
-  // FIXME Do we actually need to template conjugated_scalar
-  // on the Reference type as well as T ?
-  using reference     =
-    conjugated_scalar< /* typename Accessor::reference, */ std::complex<T>>;
   using offset_policy =
-    accessor_conjugate<typename Accessor::offset_policy, std::complex<T>>;
-
-  accessor_conjugate() = default;
-
-  accessor_conjugate(Accessor a) : acc(a) {}
-
-  reference access(pointer p, size_type i) const noexcept {
-    return reference(acc.access(p, i));
+    std::conditional_t<accessor_value_is_arith,
+		       typename Accessor::offset_policy,
+		       accessor_conjugate<typename Accessor::offset_policy>>;
+
+  accessor_conjugate(Accessor accessor) : accessor_(accessor) {}
+
+  MDSPAN_TEMPLATE_REQUIRES(
+    class OtherElementType,
+    /* requires */ (std::is_convertible_v<
+      typename default_accessor<OtherElementType>::element_type(*)[],
+      typename Accessor::element_type(*)[]
+    >)
+  )
+  accessor_conjugate(default_accessor<Accessor> accessor) : accessor_(accessor) {}
```

Comment:
(Repeating offline discussion) Your observation is correct -- `Accessor` should not go inside `default_accessor`; this code is incorrect.  The intent is conversion from nonconst element type to const element type, for the special case of `default_accessor` (since there is currently no generic way to get the const-element version of a nonconst-element accessor).

Reply:
@crtrott I fixed this and added a test.  Thanks for the catch!